diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index c574d9fb..8144ae70 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -181,6 +181,7 @@ set(HID_SRC ../third_party/hidapi/windows/hid.c)
 set(CDC_SRC hal/hal_win_kit_cdc.c)
 elseif(APPLE)
 set(CRYPTOAUTH_SRC ${CRYPTOAUTH_SRC} hal/hal_linux.c)
+set(TWI_SRC hal/hal_osx_i2c_userspace.c)
 set(HID_SRC ../third_party/hidapi/mac/hid.c)
 elseif(UNIX)
 set(CRYPTOAUTH_SRC ${CRYPTOAUTH_SRC} hal/hal_linux.c)
